src/fondant/core/component_spec.py

```diff
@@ -191,27 +163,28 @@ def tags(self) -> t.List[str]:
 
     @property
     def index(self):
-        return ComponentSubset({"fields": {}})
+        # TODO: check usage
+        return {"fields": {}}
 
     @property
-    def consumes(self) -> t.Mapping[str, ComponentSubset]:
+    def consumes(self) -> t.Mapping[str, Field]:
         """The subsets consumed by the component as an immutable mapping."""
         return types.MappingProxyType(
             {
-                name: ComponentSubset(subset)
-                for name, subset in self._specification.get("consumes", {}).items()
-                if name != "additionalSubsets"
+                name: Field(name=name, type=Type.from_json(field))
+                for name, field in self._specification["consumes"].items()
+                if name != "additionalFields"
```

Comment:
I don't think we still need this `if name != "additionalFields"`. I would implement a first version without it and then add it in a separate PR.